python/pyspark/ml/torch/distributor.py

```diff
@@ -578,19 +600,23 @@ def _run_distributed_training(
         )
         self._check_encryption()
         self.logger.info(
-            f"Started distributed training with {self.num_processes} executor proceses"
+            f"Started distributed training with {self.num_processes} executor processes"
         )
         try:
-            result = (
-                self.sc.parallelize(range(self.num_tasks), self.num_tasks)
-                .barrier()
-                .mapPartitions(spark_task_function)
-                .collect()[0]
+            rows = (
+                self.spark.range(start=0, end=self.num_tasks, step=1, numPartitions=self.num_tasks)
+                .mapInPandas(func=spark_task_function, schema="chunk binary", barrier=True)
+                .collect()
             )
+            output_bytes = b""
+            for row in rows:
+                output_bytes += row.chunk
```

Comment:
Did you test the performance of concat them ?
@HyukjinKwon Do you have better approach to concat them with better performance ?

Reply:
I test the benchmarks in https://www.guyrutenberg.com/2020/04/04/fast-bytes-concatenation-in-python/  
and `join` is the fastest

Reply:
`join` cause memory issue on master ci, I have another try with `bytearray`

```
pyspark.errors.exceptions.connect.SparkConnectGrpcException: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Java heap space"
	debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:35291 {created_time:"2023-04-04T05:17:10.693589954+00:00", grpc_status:2, grpc_message:"Java heap space"}"
```

Reply:
> join cause memory issue on master ci
What's the reason ?

Reply:
debuging, I suspect `join` will consume more memory

Reply:
But your test only runs with small dataset.

Reply:
But your test only runs with small dataset.

Reply:
That is true, it is weird since I can not repro it locally. Need future investigation.

Reply:
We can skip this for now. Use `+` instead and mark a TODO here.